scripts/rearchive-all-docs.js

```diff
@@ -0,0 +1,131 @@
+const DocArchiveManager = require('../app/js/DocArchiveManager').promises
+const MongoManager = require('../app/js/MongoManager').promises
+const { db, ObjectId } = require('../app/js/mongojs')
+const minimist = require('minimist')
+
+async function worker(projectId) {
+  try {
+    // see if the project needs to be unarchived, and unarchive it
+    const archivedDocs = await MongoManager.getArchivedProjectDocs(projectId)
+    if (archivedDocs.length) {
+      await DocArchiveManager.unArchiveAllDocs(projectId)
+    }
+
+    // get the doc content so we can validate it
+    const docs = await MongoManager.getProjectsDocs(
+      projectId,
+      { include_deleted: false },
+      { lines: 1 }
+    )
+
+    // start archiving in the background while we check the content, if it was archived to begin with
+    let archivePromise
+    if (archivedDocs.length) {
+      archivePromise = DocArchiveManager.archiveAllDocs(projectId)
+    }
+
+    let warning = false
+
+    // validate the doc contents and log any warnings to investigate later
+    for (const doc of docs) {
+      if (!doc.lines) {
+        warning = true
+        console.error('WARN:', projectId, doc._id, 'has no content')
+      }
+      // eslint-disable-next-line no-control-regex
+      if (doc.lines && doc.lines.some((line) => line.match(/(\r|\u0000)/))) {
+        warning = true
+        console.error('WARN:', projectId, doc._id, 'has invalid characters')
+      }
+    }
+
+    // ensure the archive process has finished
+    if (archivePromise) {
+      await archivePromise
+    }
```

Comment:
There's a control-flow issue here, in that `validateDocs` should have the docs before `archiveAllDocs` is called, otherwise we end up in a race.

Once we have the docs, the rest of the archive process is synchronous as it's regexing a number of docs that we already have in memory. Because of this, I wanted to
1. Fetch docs to memory for parsing
2. Call `archiveAllDocs`, so that any relevant network connections end up in-flight while we work
3. Validate the docs that are in memory

So, I think it would be better to leave this bit as-is?